### PR TITLE
Fixing hardcoded row height(to FontMetrics.getHeight()) and hardcoded black text color(to SystemColor.text)

### PR DESCRIPTION
**Pull request by grebenyukaa**
This PR attempts to fix all hardcoded row heights for JTables and JTreeTables and replaces hardcoded black text color with SystemColor.text color.

See issue https://github.com/PCGen/pcgen/issues/1760

### CODE REVIEW

**Comment by grebenyukaa** [comment]
One thing I was unable to fix is tabs/skill/SkillPointTableModel.java table with rows, colored in shades of gray. I just could not pinpoint the location, where those colors are defined.

**Comment by grebenyukaa** [comment]
Anoher thing is InfoGuidePane's subpanel, which shows loaded sources. Its caption (in_si_intro) for some reason has a fixed font size, though I can't see anything related to setting any kind of font size in pcgen/gui2/InfoGuidePane.java

**Comment by grebenyukaa** [comment]
To clarify the issue with incorrect colors in SkillPointTableModel.java:


This is how it looks by default in white on black GTK+ theme, which I intend to use, as it is the only one, which supports DPI scaling.

**Comment by grimreaper** [comment]
LGTM

File: code/src/java/pcgen/gui2/dialog/CharacterHPDialog.java [hunk 1]
```diff
@@ -115,7 +117,13 @@ public TableCellEditor getCellEditor(int row, int column)
 		table.setDefaultRenderer(JButton.class, new Renderer());
 		table.setDefaultEditor(JButton.class, new Editor());
 		table.setCellSelectionEnabled(false);
-		table.setRowHeight(new IntegerEditor(1, 10).getPreferredSize().height);
+		
+		Font curFont = table.getFont();
+		FontMetrics ftMetrics = table.getFontMetrics(curFont);
+		int ftHeight = ftMetrics.getHeight();
+		table.setRowHeight(ftHeight);
+		//table.setRowHeight(new IntegerEditor(1, 10).getPreferredSize().height);
```
Comment (grimreaper) [inline code/src/java/pcgen/gui2/dialog/CharacterHPDialog.java]: I think we an remove these comments.

**Comment by LegacyKing** [comment]
@thpr @Zaister review?

**Comment by Zaister** [comment]
Looks good to me.

**Comment by LegacyKing** [comment]
@Zaister Please merge.

**Pull request merged**